pkg/app/provider/wopi/wopi.go

```diff
@@ -142,16 +142,29 @@ func (p *wopiProvider) GetAppURL(ctx context.Context, resource *provider.Resourc
 	}
 
 	q.Add("appname", p.conf.AppName)
-	q.Add("appurl", p.appURLs["edit"][ext])
 
-	if p.conf.AppIntURL != "" {
-		q.Add("appinturl", p.conf.AppIntURL)
+	var viewAppURL string
+	if viewAppURLs, ok := p.appURLs["view"]; ok {
+		if viewAppURL, ok = viewAppURLs[ext]; ok {
+			q.Add("appviewurl", viewAppURL)
+		}
 	}
-	if viewExts, ok := p.appURLs["view"]; ok {
-		if url, ok := viewExts[ext]; ok {
-			q.Add("appviewurl", url)
+	if editAppURLs, ok := p.appURLs["edit"]; !ok {
```

Comment:
This should be `if editAppURLs, ok := p.appURLs["edit"]; ok {`

Reply:
good catch :+1: